server/src/db/migrations/20231114155041-job_processor.ts

```diff
@@ -0,0 +1,121 @@
+import { Db, MongoClient } from "mongodb";
+import { IMailingListDocument, IUploadDocument } from "shared/models/document.model";
+
+export const up = async (db: Db, _client: MongoClient) => {
+  const uploadDocumentWitJobCursor = db.collection("documents").aggregate([
+    { $match: { nom_fichier: /^(?!mailing-list)/, kind: null } },
+    {
+      $lookup: {
+        from: "jobs",
+        localField: "_id",
+        foreignField: "payload.document_id",
+        as: "jobs",
+      },
+    },
+  ]);
+
+  for await (const uploadDocumentWitJob of uploadDocumentWitJobCursor) {
+    const job = uploadDocumentWitJob.jobs.filter((j: any) => j.name === "import:document")[0] ?? null;
+    const update: Partial<IUploadDocument> = {
+      kind: "upload",
+      job_status: "pending",
+      job_error: null,
+      job_id: null,
+    };
+
+    if (job) {
+      switch (job.status) {
+        case "errored":
+          update.job_status = "error";
+          update.job_error = job.output?.error ?? "Unknown";
+          break;
+        case "finished":
+          update.job_status = "done";
+          break;
+        case "pending":
+        case "running":
+        case "will_start":
+          // Job crashed
+          update.job_status = "error";
+          update.job_error = "Unknown";
+          break;
+      }
+      update.job_id = job._id.toString();
+    } else {
+      update.job_status = uploadDocumentWitJob.import_progress === 100 ? "done" : "error";
+      update.job_error = uploadDocumentWitJob.import_progress === 100 ? null : "Unknown";
+    }
+
+    await db.collection("documents").updateOne({ _id: uploadDocumentWitJob._id }, { $set: update });
+  }
+
+  const mailingListDocumentWittJobAndMailingListCursor = db.collection("documents").aggregate([
+    { $match: { nom_fichier: /^(mailing-list)/, kind: null } },
+    { $addFields: { id: { $toString: "$_id" } } },
```

Comment:
We should consider converting `document_id` to `ObjectId` to be able to lookup directly

Reply:
And more generally all id references